traits/trait_base.py

```diff
@@ -178,18 +180,18 @@ def strx ( arg ):
 #  Constants:
 #-------------------------------------------------------------------------------
 
-StringTypes = ( str, unicode, int, long, float, complex )
+StringTypes = ( str, six.text_type, int, six.integer_types, float, complex )
 
 #-------------------------------------------------------------------------------
 #  Define a mapping of coercable types:
 #-------------------------------------------------------------------------------
 
 # Mapping of coercable types.
 CoercableTypes = {
-    long:    ( 11, long, int ),
+    six.integer_types:    ( 11, six.integer_types, int ),
```

Comment:
This doesn't look right: `six.integer_types` is a tuple, and I think we want an expanded list of types here.